bench: refactor random number generation in `stats/base/dists/studentized-range`

hrshya opened the pull request:
---
type: pre_push_report
description: Results of running various checks prior to pushing changes. report:
  - task: run_javascript_examples status: na
  - task: run_c_examples status: na
  - task: run_cpp_examples status: na
  - task: run_javascript_readme_examples status: na
  - task: run_c_benchmarks status: na
  - task: run_cpp_benchmarks status: na
  - task: run_fortran_benchmarks status: na
  - task: run_javascript_benchmarks status: na
  - task: run_julia_benchmarks status: na
  - task: run_python_benchmarks status: na
  - task: run_r_benchmarks status: na
  - task: run_javascript_tests status: na ---

Resolves #4987 

## Description

> What is the purpose of this pull request?

This pull request:

-   Fix the random number generation operation by moving it out of the benchmarking loops and initializing it beforehand to avoid interfering with the results. #4987 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   performs changes similar to #4837 #4955 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

hrshya committed:
Updated benchmarks

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js:
```diff
@@ -36,12 +36,13 @@ bench( pkg, function benchmark( b ) {
 	var y;
 	var i;
 
+	q = uniform( 0.0, 12.0 );
+	r = uniform( 2.0, 20.0 );
+	v = uniform( 2.0, 20.0 );
```
Here, `q`, `r`, and `v` should each be a `Float64Array` of length 100 and should be initialized in a loop. Reference PRs will help.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js:
```diff
@@ -36,12 +36,13 @@ bench( pkg, function benchmark( b ) {
+	q = uniform( 0.0, 12.0 );
+	r = uniform( 2.0, 20.0 );
+	v = uniform( 2.0, 20.0 );
+	y = cdf( q, r, v );
```
This should still remain in the benchmarking loop, as it is the part being benchmarked.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js:
```diff
@@ -65,11 +66,11 @@ bench( pkg+':factory', function benchmark( b ) {
 	v = 5.0;
 	r = 3.0;
 	mycdf = cdf.factory( v, r );
+	q = uniform( 0.0, 1.0 );
+	y = mycdf( q );
```
Same comment as above and applies throughout the PR.

hrshya replied:
![std](https://github.com/user-attachments/assets/ee81ab97-5fb0-43c0-97d9-1f1cb8c1afe9)
Is this the right approach ?

anandkaranubc replied:
Yes, considering you are using the correct range for random number generation.

hrshya committed:
Fixed the issues with benchmark

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

hrshya commented:
I have made the changes. Can you please review it.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js:
```diff
@@ -21,27 +21,35 @@
+	r = new Float64Array( len );
+	v = new Float64Array( len );
+	for( i = 0; i < len; i++ ) {
+		q[i] = uniform( 0.0, 12.0 );
+		r[i] = uniform( 2.0, 20.0 );
+		v[i] = uniform( 2.0, 20.0 );
```
```suggestion
		q[ i ] = uniform( 0.0, 12.0 );
		r[ i ] = uniform( 2.0, 20.0 );
		v[ i ] = uniform( 2.0, 20.0 );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js:
```diff
@@ -21,27 +21,35 @@
 var pkg = require( './../package.json' ).name;
 var cdf = require( './../lib' );
+var uniform = require('@stdlib/random/base/uniform')
 
```
```suggestion
var Float64Array = require( '@stdlib/array/float64' );
```

This applies throughout the PR

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js:
```diff
@@ -56,20 +64,25 @@ bench( pkg, function benchmark( b ) {
+	q = new Float64Array( len );
 	mycdf = cdf.factory( v, r );
+	for ( i = 0; i < len; i++ ) {
+		q[i] = uniform( 0.0, 1.0 );
```
```suggestion
		q[ i ] = uniform( 0.0, 1.0 );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js:
```diff
@@ -21,27 +21,36 @@
 	var y;
 	var i;
 
+
+	len = 100;
```
```suggestion
	len = 100;
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js:
```diff
@@ -21,27 +21,36 @@
+	r = new Float64Array( len );
+	v = new Float64Array( len );
+	for( i = 0; i < len; i++ ) {
+		p[i] = uniform( 0.0, 1.0 );
+		r[i] = uniform( 2.0, 20.0 );
+		v[i] = uniform( 2.0, 20.0 );
```
Same comment as above, and applies throughout the PR

hrshya replied:
Done.

hrshya committed:
Update lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update benchmark.js

Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js:
```diff
@@ -21,27 +21,35 @@
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pkg = require( './../package.json' ).name;
 var quantile = require( './../lib' );
+const uniform = require('@stdlib/random/base/uniform');
```
```suggestion
var uniform = require('@stdlib/random/base/uniform');
```

anandkaranubc reviewed (changes requested):
Just some minor changes; the rest of the PR looks ready for merge. Great work!

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js:
```diff
@@ -21,27 +21,35 @@
 var pkg = require( './../package.json' ).name;
 var quantile = require( './../lib' );
+const uniform = require('@stdlib/random/base/uniform');
 
```
```suggestion
var Float64Array = require( '@stdlib/array/float64' );

```

anandkaranubc replied:
You need to add this line in order to use `Float64Array`

hrshya committed:
Update lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya commented:
> Just some minor changes; the rest of the PR looks ready for merge. Great work!

Thank you, Can you please review it.

hrshya commented:
Hi @anandkaranubc , Can you pls review it.

anandkaranubc commented:
> Hi @anandkaranubc , Can you pls review it.

On it!

anandkaranubc commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js:
```diff
@@ -21,27 +21,35 @@
+var uniform = require( '@stdlib/random/base/uniform' );
+var Float64Array = require( '@stdlib/array/float64' );
 var pkg = require( './../package.json' ).name;
 var cdf = require( './../lib' );
 
```
```suggestion
var cdf = require( './../lib' );


```

Header comments should be preceded by two empty lines.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js:
```diff
@@ -21,27 +21,36 @@
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pkg = require( './../package.json' ).name;
 var quantile = require( './../lib' );
+var uniform = require('@stdlib/random/base/uniform');
```
```suggestion
var uniform = require( '@stdlib/random/base/uniform' );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js:
```diff
@@ -21,27 +21,36 @@
 
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pkg = require( './../package.json' ).name;
 var quantile = require( './../lib' );
+var uniform = require('@stdlib/random/base/uniform');
+var Float64Array = require( '@stdlib/array/float64' );
```
```suggestion
var isnan = require( '@stdlib/math/base/assert/is-nan' );
var uniform = require( '@stdlib/random/base/uniform' );
var Float64Array = require( '@stdlib/array/float64' );
var pkg = require( './../package.json' ).name;
var quantile = require( './../lib' );
```

Just some rearrangement for consistency.

anandkaranubc reviewed (changes requested):
Just some minor changes. Rest, the PR looks good!

hrshya committed:
Update lib/node_modules/@stdlib/stats/base/dists/studentized-range/cdf/benchmark/benchmark.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/stats/base/dists/studentized-range/quantile/benchmark/benchmark.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

anandkaranubc commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

anandkaranubc commented:
@hrshya Can you also change the title to:

bench: refactor random number generation in `stats/base/dists/studentized-range`

Make sure you apply backticks around stats/base/dists/studentized-range

hrshya commented:
@anandkaranubc Done!

anandkaranubc reviewed (approved):
Great work! Time to wait for a maintainer to merge this PR :)

Planeshifter commented:
Thanks @hrshya for PR and @anandkaranubc for review; much appreciated!
